pkg/cluster/cluster_config.go

```diff
@@ -67,32 +66,28 @@ func GetClusterServiceVersion(ctx context.Context, c client.Client, watchNameSpa
 
 type Platform string
 
-// isSelfManaged checks presence of ClusterServiceVersions:
-// when CSV displayname contains OpenDataHub, return 'OpenDataHub,nil' => high priority
-// when CSV displayname contains SelfManagedRhods, return 'SelfManagedRhods,nil'
-// when in dev mode and  could not find CSV (deploy by olm), return "", nil
-// otherwise return "",err.
-func isSelfManaged(cli client.Client) (Platform, error) {
-	clusterCsvs := &ofapi.ClusterServiceVersionList{}
-	err := cli.List(context.TODO(), clusterCsvs)
-	if err != nil {
-		return "", err
-	} else { //nolint:golint,revive // Readability on else
-		for _, csv := range clusterCsvs.Items {
-			if strings.Contains(csv.Spec.DisplayName, string(OpenDataHub)) {
-				return OpenDataHub, nil
-			}
-			if strings.Contains(csv.Spec.DisplayName, string(SelfManagedRhods)) {
-				return SelfManagedRhods, nil
-			}
+// detectSelfManaged detects if it is Self Managed Rhods or OpenDataHub.
+func detectSelfManaged(cli client.Client) (Platform, error) {
+	variants := map[string]Platform{
+		"opendatahub-operator": OpenDataHub,
+		"rhods-operator":       SelfManagedRhods,
+	}
+
+	for k, v := range variants {
+		exists, err := OperatorExists(cli, k)
+		if err != nil {
+			return Unknown, err
+		}
+		if exists {
+			return v, nil
 		}
 	}
 
 	return Unknown, nil
 }
 
-// isManagedRHODS checks if CRD add-on exists and contains string ManagedRhods.
-func isManagedRHODS(cli client.Client) (Platform, error) {
+// detectManagedRHODS checks if CRD add-on exists and contains string ManagedRhods.
+func detectManagedRHODS(cli client.Client) (Platform, error) {
```

Comment:
can we do like this 
```
if cs.Name == string("addon-managed-odh-catalog") {
				return ManagedRhods, nil
			}
```
then in 
```
const (
	// ManagedRhods defines expected addon catalogsource.
	ManagedRhods Platform = "Managed Red Hat OpenShift AI Managed"
	// SelfManagedRhods defines display name in csv.
	SelfManagedRhods Platform = "Self Managed Red Hat OpenShift AI"
	// OpenDataHub defines display name in csv.
	OpenDataHub Platform = "Open Data Hub Operator"
	// Unknown indicates that operator is not deployed using OLM.
	Unknown Platform = ""
)
```

Reply:
Pushed. But probably it's better to have that changes (last two) as a separate PR(s) due to squashing policy. They are not related to the original issue.

Reply:
> Pushed. But probably it's better to have that changes (last two) as a separate PR(s) due to squashing policy. They are not related to the original issue.

done on top on this PR

Reply:
sure, just bring this up not necessary be in the same PR